server/controllers/notifications.js

```diff
@@ -0,0 +1,82 @@
+const Notification = require("../models/Notifications");
+const asyncHandler = require("express-async-handler");
+const mongoose = require('mongoose');
+const { ObjectId } = mongoose.Types;
+
+// @route POST /notification
+// @desc create a new notification
+// @access Private
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  try {
+    const userId = req.user.id;
+    
+    const { type, date, title, description } = req.body;
+    const notificatonModel = {
+        userId: ObjectId(userId),
+        date: new Date(date),
+        type,
+        title,
+        description,
+    };
+    
+    const notification = await Notification.create(notificatonModel); 
+    res.status(201).json({
+      success: { notification }
+    });
+  } catch (error) {
+    next(error);
+  } 
+});
+
+// @route GET /notification/all
+// @desc list of notifications; only get read if query provided 
+// @access Private
+exports.notificationList = asyncHandler(async (req, res, next) => {
+  try {
+    const userId = req.user.id;
+    const { unread } = req.query;
+    
+    const options = { userId: ObjectId(userId) };
+    if (unread == 'true') options.read = true;
```

Comment:
Prefer strict equality `===` OR simply `if (unread)`

Reply:
hi @ezeilo-su. Thanks for the feedback. I think === would work better because the  query `unread` has to be equal to  `true`.  If its something else, like ?unread=foo, it should not set the `options.read` to true.